### PR TITLE
restrict proposals to 512 bytes

### PR DESCRIPTION
**Pull request by nmarley**
Related to https://github.com/dashpay/dash/pull/2004 , this is needed for validation rules to stay in sync between Sentinel and Core.

### CODE REVIEW

File: test/unit/models/test_proposals.py [hunk 1]
```diff
@@ -277,3 +277,13 @@ def test_approved_and_ranked(go_list_proposals):
...
     assert prop_list[1].object_hash == u'0523445762025b2e01a2cd34f1d10f4816cf26ee1796167e5b029901e5873630'
+
+
+def test_proposal_size(proposal):
+    orig = Proposal(**proposal.get_dict())  # make a copy
+
+    # fixture as-is should be valid
+    assert proposal.is_valid() is True
+
+    proposal.url = 'https://' + ('x' * 513) + '.com/'
```
Comment (UdjinM6) [inline test/unit/models/test_proposals.py]: This looks a bit unrelated tbh :) `'x' * 512` would also pass the assert below, no?

Comment (nmarley) [reply test/unit/models/test_proposals.py]: Yes. The intent was to ensure the total proposal size is over the limit.

I'm not sure what's not related -- what would you suggest changing?

Comment (UdjinM6) [reply test/unit/models/test_proposals.py]: Shouldn't it test smth like:
- 512 bytes proposal - ok
- 513 bytes proposal - fail

The way it's written now there is no explicit test for that 512 (MAX_DATA_SIZE) bytes threshold.

Comment (nmarley) [reply test/unit/models/test_proposals.py]: Ok, so you wanna tighten the test then. I agree, will take just a bit more work to setup mock object.

Comment (nmarley) [reply test/unit/models/test_proposals.py]: Good catch! Tightening the test caught a bug where `>=` should have been `>`.

**Review by UdjinM6 — APPROVED** [review]
utACK 👍
